src/backend/distributed/worker/worker_shard_visibility.c

```diff
@@ -112,10 +113,38 @@ ErrorIfRelationIsAKnownShard(Oid relationId)
 	}
 
 	const char *relationName = get_rel_name(relationId);
+
 	ereport(ERROR, (errmsg("relation \"%s\" is a shard relation ", relationName)));
 }
 
 
+/*
+ * ErrorIfChangingKnownShard errors out if the relation with relationId is
+ * a known shard and manual changes on known shards are disabled. This is
+ * valid for only non-citus (external) connections.
+ */
+void
+ErrorIfChangingKnownShard(Oid relationId)
+{
+	if (IsCitusInitiatedRemoteBackend() || EnableManualChangesToShards)
+	{
+		return;
+	}
+
+	/* search the relation in all schemas */
+	bool onlySearchPath = false;
+	if (RelationIsAKnownShard(relationId, onlySearchPath))
+	{
+		const char *relationName = get_rel_name(relationId);
+		ereport(ERROR, (errmsg("relation \"%s\" is a shard relation ", relationName),
+						errdetail("You should not change shards directly, but use "
+								  "the distributed table instead"),
+						errhint("To modify shards directly, "
+								"set citus.enable_manual_changes_to_shards to on")));
```

Comment:
Looking at errdetail and errhint together now, they partially overlap.

Maybe we remove the errdetail and use: errhint("Use the distributed table or set citus.enable_manual_changes_to_shards to on to modify shards directly")